media/au.ksy

```diff
@@ -0,0 +1,61 @@
+meta:
+  id: au
+  title: AU audio format
+  file-extension:
+    - au
+    - snd
+  xref:
+    justsolve:
+      - AU
+    mime:
+      - audio/basic
+    pronom:
+      - x-fmt/139
+    wikidata:
+      - Q672985
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+doc: |
+  Test files for AU can be found in any Python release in the
+  directory Lib/test/sndhdrdata/
+doc-ref: https://en.wikipedia.org/wiki/Au_file_format
+doc-ref: https://web.archive.org/web/20121028010008/http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/AU/AU.html
+seq:
+  - id: magic
+    contents: ".snd"
+  - id: header_size
```

Comment:
I don't see it used in the spec.

Reply:
It is named differently in the different specs. In this one it is "hdr_size:

https://pubs.opengroup.org/external/auformat.html

In this one it is "offset to data":

https://web.archive.org/web/20121028010008/http://www-mmsp.ece.mcgill.ca/Documents/AudioFormats/AU/AU.html

because the audio data immediately follows the header.

Reply:
Immediately follows the header of the specific size. The header must be of that size then. It is done intentionally for forward compatibility.

Reply:
Very likely. In practice the header always seems to have been 24 though and the format is no longer in widespread use. One check that could be added is that the value of that field is at least 24.

Reply:
>One check that could be added is that the value of that field is at least 24.

It violates the goal of forward comatibility.

Reply:
You are doing it wrong.

```yaml
types:
  preheader:
    seq:
       - id: magic
         contents: ".snd"
       - id: header_size
         type: u4
  header:
    seq:
      - id: preheadet
        type: preheader
     - id: rest
       type: rest_of_header
       size: preheader.header_size - sizeof<preheader>
```

Reply:
If there were lots of different variations of the format I would have agreed with you. But this format hasn't changed in over 30 years and has effectively been abandoned by the industry. There has been NO file that has anything else in that header except the value 24. So why try to make it work for variants that don't exist and that won't exist?

But you are right, I am doing it wrong. I should not say that the minimum value is 24, but that the value equals 24 :-)

Reply:
> If there were lots of different variations of the format I would have agreed with you. But this format hasn't changed in over 30 years and has effectively been abandoned by the industry. There has been NO file that has anything else in that header except the value 24. So why try to make it work for variants that don't exist and that won't exist?
> 
> But you are right, I am doing it wrong. I should not say that the minimum value is 24, but that the value equals 24 :-)

I am taking this back. After some more searching I found a header with size 40. I will change it as suggested. Thank you!

Reply:
I went through the specification again. Everything that is part of the header but after byte 24 is part of a description or textual information (ASCII encoded). I have reworked the header to reflect this. I have kept the "min: 24" statement as it is a hard restriction that directly follows from the specification. It absolutely makes no sense to make it more generic.

Reply:
As said, I found other files with a different header, so this has been fixed.